File: libs/cosmology.cpp
#include "cosmology.h"
#include "const.h"

static double omega_m= 0.3;  // Omega_m(z=0)

void cosmology_set_omega_m(const double omega_m_)
{
  omega_m= omega_m_;
}

double cosmology_omega_m()
{
  return omega_m;
}

double cosmology_rho_m()
{
  // Mean matter density at z=0, or comoving mean matter density
  return omega_m*rho_crit_0;
}

File: py/mockgallib/cosmology.py
import mockgallib._mockgallib as c


def set(omega_m):
    """Set omega_m
    Args:
        omega_m: cosmological matter density Omega_m(z=0)
    """
    
    c._cosmology_set(omega_m)


def compute_comoving_distance(a):
    """Computes comoving distance correcspons to scale factor a"""
    
    return c._cosmology_compute_comoving_distance(a)


def rhom():
    """Mean matter density at z=0 in [M_solar (1/h Mpc)^-3]"""
    return c._cosmology_rhom()
